File: benchmarks/go/09_i64_hotloop.go
// Fixed-Width i64 Arithmetic Hot Loop Benchmark - Go Implementation

package main

import "fmt"

func main() {
	const n int64 = 2000000
	var acc int64 = 1

	for i := int64(1); i <= n; i++ {
		acc += i
		acc -= i % 7
		acc += (i * 3) % 11
		if i%5 == 0 {
			acc = acc/2 + 17
		}
	}

	fmt.Printf("i64 hotloop checksum: %d\n", acc)
}

File: benchmarks/python/09_i64_hotloop.py
def main():
    n = 2000000
    acc = 1
    i = 1
    while i <= n:
        acc += i
        acc -= i % 7
        acc += (i * 3) % 11
        if i % 5 == 0:
            acc = acc // 2 + 17
        i += 1
    print(acc)


if __name__ == "__main__":
    main()
